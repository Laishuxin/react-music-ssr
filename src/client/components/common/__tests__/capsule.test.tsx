import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import Capsule from '../capsule'

describe('capsule', () => {
  const fn = jest.fn()
  it('should render the correct capsule', () => {
    const { getByTestId } = render(
      <Capsule data-testid='test-capsule' text='test' />,
    )
    const { getByTestId: getByTestId2 } = render(
      <Capsule
        onClick={fn}
        data-testid='test-capsule2'
        tagName='button'
        text='test'
      />,
    )
    const element = getByTestId('test-capsule')
    const element2 = getByTestId2('test-capsule2')
    expect(element).toMatchSnapshot('capsule-default')

    expect(element2.tagName.toUpperCase()).toBe('BUTTON')
    expect(element2).toMatchSnapshot()

    fireEvent.click(element2)
    expect(fn).toHaveBeenCalled()
  })
})
