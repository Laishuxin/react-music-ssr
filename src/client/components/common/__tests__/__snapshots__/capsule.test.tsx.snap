// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`capsule should render the correct capsule 1`] = `
<button
  class="capsule rounded-lg p-1 text-current border-2 border-current border-solid"
  data-testid="test-capsule2"
>
  test
</button>
`;

exports[`capsule should render the correct capsule: capsule-default 1`] = `
<span
  class="capsule rounded-lg p-1 text-current border-2 border-current border-solid"
  data-testid="test-capsule"
>
  test
</span>
`;
